synapse/lib/stormlib/infosec.py

```diff
@@ -409,6 +409,72 @@ def CVSS3_1_calc(vdict):
 def roundup(x):
     return (math.ceil(x * 10) / 10.0)
 
+@s_stormtypes.registry.registerLib
+class MitreAttackFlowLib(s_stormtypes.Lib):
+    '''
+    A Storm library which implements modeling MITRE ATT&CK Flow diagrams.
+    '''
+    _storm_lib_path = ('infosec', 'mitre', 'attack', 'flow')
+    _storm_locals = (
+        {'name': 'ingest', 'desc': 'Ingest a MITRE ATT&CK Flow diagram in JSON format.',
+         'type': {'type': 'function', '_funcname': '_storm_query',
+                  'args': (
+                      {'name': 'flow', 'type': 'data', 'desc': 'The JSON data to ingest.'},
+                  ),
+                  'returns': {'type': 'null'}}},
+
+    )
+    _storm_query = '''
+        function ingest(flow) {
+            $lib.cast(it:mitre:attack:flow:json, $flow)
+
+            $guid = $lib.guid($flow)
+
+            $objs_byid = ({})
+            $objs_bytype = ({})
+
+            for $obj in $flow.objects {
+                $id = $obj.id
+                $objs_byid.$id = $obj
+
+                $type = $obj.type
+                if (not $objs_bytype.$type) {
+                    $objs_bytype.$type = ([])
+                }
+                $objs_bytype.$type.append($obj)
+            }
+
+            $attack_flow = $objs_bytype."attack-flow".0
```

Comment:
Is it enforced by the schema that this will always be present with len>0?

Reply:
Yep! Not just len>0, but exactly one in the document.